packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -720,7 +722,15 @@ class _AppBarState extends State<AppBar> {
 
     final bool hasDrawer = scaffold?.hasDrawer ?? false;
     final bool hasEndDrawer = scaffold?.hasEndDrawer ?? false;
-    final bool canPop = parentRoute?.canPop ?? false;
+    if (parentRoute == null) {
+      previouslyImplyLeadingPopButton = false;
```

Comment:
I thought the objective was to leave the back button alone if a route became inactive (it was popped) and it had a back button before it became inactive? Maybe this should be?

```dart
previouslyImplyLeadingPopButton ??= false;
```